src/workerd/api/crypto/dh.c++

```diff
@@ -40,80 +36,88 @@ BIGNUM* (*FindDiffieHellmanGroup(const char* name))(BIGNUM*) {
 
   return nullptr;
 }
-} // namespace
-
-jsg::Ref<CryptoImpl::DiffieHellmanHandle> CryptoImpl::DiffieHellmanGroupHandle(kj::String name) {
-  return jsg::alloc<DiffieHellmanHandle>(name);
-}
-
-jsg::Ref<CryptoImpl::DiffieHellmanHandle> CryptoImpl::DiffieHellmanHandle::constructor(
-    jsg::Lock &js, kj::OneOf<kj::Array<kj::byte>, int> sizeOrKey,
-    kj::OneOf<kj::Array<kj::byte>, int> generator) {
-  return jsg::alloc<DiffieHellmanHandle>(sizeOrKey, generator);
-}
-
-bool CryptoImpl::DiffieHellmanHandle::VerifyContext() {
-  int codes;
-  if (!DH_check(dh, &codes))
-    return false;
-  verifyError = codes;
-  return true;
-}
-
-CryptoImpl::DiffieHellmanHandle::DiffieHellmanHandle(
-    kj::OneOf<kj::Array<kj::byte>, int> &sizeOrKey, kj::OneOf<kj::Array<kj::byte>,
-    int> &generator) : verifyError(0) {
-  JSG_REQUIRE(Init(sizeOrKey, generator), Error, "DiffieHellman init failed");
-}
 
-CryptoImpl::DiffieHellmanHandle::DiffieHellmanHandle(kj::String& name) : verifyError(0) {
-  JSG_REQUIRE(InitGroup(name), Error, "DiffieHellman init failed");
-}
-
-bool CryptoImpl::DiffieHellmanHandle::InitGroup(kj::String& name) {
-  auto group = FindDiffieHellmanGroup(name.begin());
+kj::Own<DH> initDhGroup(kj::StringPtr name) {
+  auto group = findDiffieHellmanGroup(name.begin());
   JSG_REQUIRE(group != nullptr, Error, "Failed to init DiffieHellmanGroup: invalid group. Only "
               "groups {modp14, modp15, modp16, modp17, modp18} are supported.");
```

Comment:
Do we want to consolidate this list with the list in findDiffieHellman? May not be necessary if the list is unlikely to change

Reply:
not critical right now